src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/ClassicalControl.qs

```diff
@@ -909,4 +909,95 @@ namespace Microsoft.Quantum.Testing.ClassicalControl {
             Bar(r1);
         }
     }
+}
+
+// =================================
+
+// Inequality with ApplyConditionally
+namespace Microsoft.Quantum.Testing.ClassicalControl {
+    open SubOps;
+
+    operation Bar(r : Result) : Unit { }
+
+    operation Foo() : Unit {
+        let r1 = Zero;
+        let r2 = Zero;
+
+        if (r1 != r2) {
+            Bar(r1);
+        }
+        else {
+            SubOp1();
+        }
+    }
+}
+
+// =================================
+
+// Inequality with Apply If One Else Zero
+namespace Microsoft.Quantum.Testing.ClassicalControl {
+    open SubOps;
+
+    operation Bar(r : Result) : Unit { }
+
+    operation Foo() : Unit {
+        let r = Zero;
+
+        if (r != Zero) {

```

Comment:
In this case (and the other Apply If cases) you flip the constant, correct? (such that this becomes "if (r == One) { Bar(r); } else { SubOp1(); }")

Reply:
Yes, that is correct.